doc/sphinx/proofs/writing-proofs/rewriting.rst

```diff
@@ -384,91 +381,104 @@ Rewriting with definitional equality
            exact H.
          Qed.
 
-.. _performingcomputations:
+.. _applyingconversionrules:
 
-Performing computations
----------------------------
+Applying conversion rules
+-------------------------
+
+These tactics apply reductions and expansions, replacing :term:`convertible` terms
+with equivalent terms.
+They implement different specialized uses of the
+:tacn:`change` tactic.  Other ways to apply these reductions are through
+the :cmd:`Eval` command, the `Eval` clause in the :cmd:`Definition`/:cmd:`Example`
+command and the :tacn:`eval` tactic.
+
+Conversions (except, in some cases, WHICH ARE? for :tacn:`vm_compute` and
```

Comment:
AFAIK calling `vm_compute` or `native_compute` on the goal will introduce an explicit cast in the proof term (https://coq.inria.fr/refman/language/core/definitions.html#grammar-token-term_cast) so that the kernel knows which reduction machine to use.

Reply:
That also matches my understanding.

Reply:
OK. thanks.  I suppose it's possible for those 2 tactics to reverse the cast after the reductions are complete, but I don't see why anyone would care.

Reply:
>  I suppose it's possible for those 2 tactics to reverse the cast after the reductions are complete, but I don't see why anyone would care.

It's actually very important that these tactics leave behind cast nodes, and ideally all of the conversion tactics would do the same.  If there's no evidence in the proof term, then the kernel doesn't know what reduction/conversion strategy to use at `Qed`-time, and the `Qed` can be extremely slow.  See also https://github.com/coq/coq/issues/4637